src/smbclient/_os.py

```diff
@@ -1019,6 +1020,36 @@ def setxattr(path, attribute, value, flags=0, follow_symlinks=True, **kwargs):
         set_info(transaction, ea_info)
 
 
+def liststreams(path, follow_symlinks=True, **kwargs) -> list[str]:
```

Comment:
Using `list[str]` will require the future import at the top of the file for older Python versions.

```python
from __future__ import annotations
```

While we haven't typed any of the other functions yet it would be great to start doing it for new additions

```suggestion
def liststreams(
    path: str,
    follow_symlinks: bool = True,
    **kwargs: typing.Any,
) -> list[str]:
```

Reply:
I'm also midway through adding proper inferred types for `open_file`, though the buffering needing a "nonzero" case is causing me grief. Hope to finish shortly.